22_pipeslab/pipeslab.txt

```diff
@@ -1,34 +1,62 @@
-For each of the following, type a single command line that performs:
-
-  1. Print the number of lines from "ls -l" which represent a path that has execute permission to their owner.
-  Answer: 
-
-  2. Print the number of lines from "ls -l" which represent a path that has execute permission to their owner, group and others.
-  Answer: 
-
-  3. Print the number of lines from "ls -l" which represent paths that were modified today (no matter which day we run the command)
-  Answer: 
-
-  4. Modify (1) and (2) to count files only (no directories)
-  Answer:
-
-  5. The command /sbin/ifconfig shows information regarding the machine's network interfaces. Use the command to print only the IP address of the machine. Remember not to show the special loopback address 127.0.0.1
-
-6. Joe wanted to sort a text file in place so he tried to run the command "sort myfile.txt > myfile.txt". What happened to the file? Why?
-
-7. The command "who" shows all connected users along with the connection start date. For example running it on my machine gives:
-
-$ who
-ynonperek console  Nov 18 16:29 
-ynonperek ttys000  Dec 10 11:58 
-ynonperek ttys001  Nov 20 08:35 
-ynonperek ttys005  Dec 10 17:04 
-
-I want to show how many connections began in each month so I tried to run the command:
-$ who | cut -d ' ' -f 3 | sort | uniq -c
-
-What was printed? Why? Fix the command line to get the output:
-2 Dec
-2 Nov
-
-
+For each of the following, type a single command line that performs:
+
+
+
+  1. Print the number of lines from "ls -l" which represent a path that has execute permission to their owner.
+
+  Answer: ls -l | cut -c 4 | grep -c x 
+
+
+  2. Print the number of lines from "ls -l" which represent a path that has execute permission to their owner, group and others.
+
+  Answer:ls -l | cut -c 4,7,10 | grep -c x 
+
+
+
+  3. Print the number of lines from "ls -l" which represent paths that were modified today (no matter which day we run the command)
+
+  Answer: ls -l  cut -f 6,7-d' ' |grep +(date +%b� �%d) wc -l
+
+
+
+  4. Modify (1) and (2) to count files only (no directories)
+
+  Answer:ls -l | cut -f1 | grep -c x
+
+
+
+  5. The command /sbin/ifconfig shows information regarding the machine's network interfaces. Use the command to print only the IP address of the machine. Remember not to show the special loopback address 127.0.0.1
+
+    /sbin/ifconfig | cut ...
+
+6. Joe wanted to sort a text file in place so he tried to run the command "sort myfile.txt > myfile.txt". What happened to the file? Why?
+
+   the file myfile.txt was deleted ( run over..)

```

Comment:
Yes. And it's a bit surprising because one would hope the output of sort will be written into the file, and the result will have that file sorted.

Unfortunately the order of execution in a shell pipeline begins from the end, and so what actually happens:

1. Bash opens the file myfile.txt for writing, truncating its contents
2. Bash runs sort myfile.txt which tries to sort the lines in the file, but the file is already empty so nothing to sort
3. The output of sort is written back to the file, leaving it empty